src/test/ui/consts/const-eval/ub-ref.rs

```diff
@@ -1,12 +1,9 @@
-// ignore-tidy-linelength
 #![feature(const_transmute)]
 #![allow(const_err)] // make sure we cannot allow away the errors tested here
 
 use std::mem;
 
-const UNALIGNED: &u16 = unsafe { mem::transmute(&[0u8; 4]) };
-//~^ ERROR it is undefined behavior to use this value
-//~^^ type validation failed: encountered unaligned reference (required 2 byte alignment but found 1)
+const UNALIGNED: &u16 = unsafe { mem::transmute(&[0u8; 4]) }; // Ok (CTFE does not check alignment)
```

Comment:
Could this test be adjusted to make use of the value and error there at least?

Reply:
Why would that be better? Seems more noisy, mostly.

Reply:
Mostly just to make sure that a use of the value would be caught (i.e. I'm unsure about test coverage).

Reply:
IIRC we do have several tests elsewhere that make sure error'd constants are unusable (and also in the current implementation CTFE literally does not return a value that it could even use). @oli-obk probably knows where.

I have mixed feelings about testing a precise thing here vs expanding our testing envelope by basically a "random walk".

Reply:
imo there's no reason const eval should even error on dereferencing unaligned things. If you do UB in const eval, that's your fault and we don't promise to catch it if it's expensive to do so. We have miri to catch UB.

There's already so many things const eval doesn't catch, missing alignment checks isn't important enough to be one of the things we do catch.

Reply:
Oh I just realized this is not even the test I was thinking we would be talking about.

@Centril with this patch, using that value is indeed not caught any more. What I could do (I think that would not be too messy) is that const *validation* checks for alignment even though during const evaluation, alignment is ignored. I would add a `check_ptr_access_align` or so that "overrides" the boolean in the machine trait.